ecosystem/gstreamer_plugin/gst_mtl_st20p_rx.c

```diff
@@ -461,25 +498,40 @@ static GstFlowReturn gst_mtl_st20p_rx_create(GstBaseSrc* basesrc, guint64 offset
 
   if (!frame) {
     GST_INFO("Failed to get frame EOS");
-    GST_OBJECT_UNLOCK(src);
     return GST_FLOW_EOS;
   }
 
-  gst_buffer_map(buf, &dest_info, GST_MAP_WRITE);
+  gst_buffer_map(*buffer, &dest_info, GST_MAP_WRITE);
 
-  fill_size = gst_buffer_fill(buf, 0, frame->addr[0], src->frame_size);
-  GST_BUFFER_PTS(buf) = frame->timestamp;
+  gsize fill_size = gst_buffer_fill(*buffer, 0, frame->addr[0], src->frame_size);
+  GST_BUFFER_PTS(*buffer) = frame->timestamp;
```

Comment:
declarations should be done up top of the function

Reply:
Fixed in bedd0895579ff54984a2df2d2aed7191e446539a